### PR TITLE
feat(vscodium): VSCodium / Open VSX support with guided installation (v0.83.0)

### PR DESCRIPTION
**Pull request by Shen-Ming-Hong**
## Summary

Adds VSCodium / Open VSX support by removing the hard `extensionDependencies` requirement for `platformio.platformio-ide` and replacing it with a guided provider-installation flow.

This update also makes PlatformIO readiness depend on a successfully executable Core invocation, so a present-but-blocked `pio.exe` no longer causes a false unavailable or false-ready state.

---

### Guided provider installation

- Keeps the existing all-board setup behavior when the Blockly editor is opened.
- Tries `platformio.platformio-ide`, then `pioarduino.pioarduino-ide` when the first provider is unavailable.
- Opens the Extensions search when both provider installations fail.
- Only offers reload after a provider installation command succeeds; a failed installation is never reported as ready.
- Removes long-running penv polling and path-existence-based readiness assumptions.
- Distinguishes an installed provider that is still initializing from a completely missing environment.

### Resilient PlatformIO Core resolution

- Resolves and probes actual executable invocations instead of treating file existence as readiness.
- Checks `platformio-ide.customPATH`, `PLATFORMIO_CORE_DIR`, the Windows system-drive Core directory, the default user Core directory, and `PATH`.
- Prefers `SystemDrive` for Windows system-drive Core discovery even when the user profile is on another drive.
- Continues after a found launcher fails to execute.
- Falls back to the Core environment's `python -m platformio` invocation when the direct launcher is blocked.
- Reuses the resolved invocation for device detection, compile, upload, PlatformIO diagnostics, and Arduino Serial Monitor.
- Distinguishes "not found" from "found but not executable" in diagnostics and upload guidance.

### Serial Monitor compatibility

- Removes the unrelated fixed-penv guard from the CyberBrick Serial Monitor.
- Preserves custom Python/pyserial operation.
- Uses the existing `mpremote` backend when USB detection succeeds through `mpremote` or Python monitor startup fails.
- Quotes terminal arguments explicitly on PowerShell and POSIX shells.

### Security and maintainability

- Uses `execFile` and `spawn(..., { shell: false })` for production PlatformIO command execution.
- Centralizes PlatformIO candidate discovery, probing, and fallback behavior in `PlatformioInvocationResolver`.
- Adds regression coverage for blocked Windows launchers, cross-drive/non-ASCII user paths, custom Core directories, false-ready installation states, and Serial Monitor fallback behavior.

---

## Related spec

- Spec: `specs/063-vscodium-openvsx-guided-install/spec.md`
- Plan: `specs/063-vscodium-openvsx-guided-install/plan.md`
- Tasks: `specs/063-vscodium-openvsx-guided-install/tasks.md`

---

## Verification

- ✅ `npm run lint`
- ✅ `npm run compile`
- ✅ `npm run validate:i18n` — 14/14 non-base locales passed
- ✅ Codex review edge-case tests: 37/37 passed
- ✅ Full unit suite: 936 passing, 1 pending
- ⚠️ 1 pre-existing unrelated failure remains in `MicropythonUploader CyberBrick helper commands`: `wraps rc_main.py with a single OTA startup call without secrets`
- ✅ Test VSIX 0.83.0 was produced and validated before this PR update

---

## Review focus

- Windows `pio.exe` present but blocked, with `python.exe -m platformio` available
- `PLATFORMIO_CORE_DIR`, `SystemDrive`, and cross-drive/non-ASCII Windows user paths
- No reload/ready message after both provider installation attempts fail
- Provider-installed/Core-pending guidance does not point back to an ineffective install flow
- Arduino compile/upload/monitor reuse the resolved invocation
- CyberBrick custom Python and `mpremote` Serial Monitor behavior remains intact

---

Supersedes #91 and retains its settings guard and guided-installation changes.

### CODE REVIEW

**Comment by Shen-Ming-Hong** [comment]
@codex review

File: src/services/platformioInvocationResolver.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	return platform === 'win32'
+		? pathApi.join(coreDir, 'penv', 'Scripts')
+		: pathApi.join(coreDir, 'penv', 'bin');
+}
+
+function getWindowsSystemDriveCoreDir(
+	homeDir: string,
+	env: NodeJS.ProcessEnv
+): string | null {
+	const driveRoot = path.win32.parse(homeDir).root || (env['SystemDrive'] ? `${env['SystemDrive']}\\` : '');
```
[the remark this answers is not in the document]
Comment (Shen-Ming-Hong) [reply src/services/platformioInvocationResolver.ts]: 已在 eae90d7 修正：Windows Core 候選現在優先採用 SystemDrive，缺少時才 fallback 到使用者目錄磁碟；並新增 homeDir=D:、SystemDrive=C: 的跨磁碟回歸測試。

File: src/services/arduinoUploader.ts [hunk 2]
```diff
@@ -1039,8 +1122,12 @@ export class ArduinoUploader {
 			sendProgress('checking_pio', 15, 'Checking compiler...');
 			const hasPio = await this.checkPioInstalled();
 			if (!hasPio) {
-				sendProgress('failed', 15, 'PlatformIO not found', undefined, 'PIO_NOT_FOUND');
-				return this.createFailureResult(this.startTime, 'none', 'checking_pio', 'PlatformIO CLI not found');
+				const foundButUnavailable = (this.lastPioResolution?.foundCandidates.length ?? 0) > 0;
+				const notReady = foundButUnavailable
+					? 'PlatformIO Core was found but could not be started. Check the PlatformIO diagnostic panel for details.'
+					: 'PlatformIO environment not found. Open the Blockly editor to trigger automatic setup.';
```
[the remark this answers is not in the document]
Comment (Shen-Ming-Hong) [reply src/services/arduinoUploader.ts]: 已在 eae90d7 修正：當沒有 Core 候選但 provider 已安裝時，改為顯示 Core 尚在初始化／尚未就緒，不再引導重新開啟 Blockly；並新增 provider-installed/Core-missing 測試。

**Comment by Shen-Ming-Hong** [comment]
@codex review

**Comment by Shen-Ming-Hong** [comment]
@codex review

Addressed both P2 findings in eae90d7 with regression tests. Please review the latest commit.

**Comment by Shen-Ming-Hong** [comment]
@codex review

GitHub account linking is now enabled. Please review the latest commit eae90d7.

**Comment by Shen-Ming-Hong** [comment]
@codex

**Comment by Shen-Ming-Hong** [comment]
@codex

**Comment by Shen-Ming-Hong** [comment]
@codex

**Comment by Shen-Ming-Hong** [comment]
@codex

**Comment by Shen-Ming-Hong** [comment]
@codex
